pkg/datapath/iptables/iptables.go

```diff
@@ -1140,3 +1151,25 @@ func (m *IptablesManager) addCiliumNoTrackXfrmRules() error {
 	}
 	return nil
 }
+
+func (m *IptablesManager) addCiliumENIRules() error {
```

Comment:
My next question is, what's our plan for a non-iptables version of this functionality? :-)

Reply:
No plan at the moment. I'm mostly focusing on getting a fix so NodePort works as expected on EKS, I don't know if we want or need a non-iptables version. It's only two rules per interface used between the nodes, so shouldn't be adding to much overhead, but maybe finding an alternative is desirable... What is your opinion on the topic?

Reply:
We're aiming to provide an entirely iptables-free implementation, so that's where my question comes from. I think it's very reasonable to have this to solve the functional question first, then we can address this question later. This is mostly fishing for thoughts :-)

Reply:
Just got bitten by this, thankfully after 2h of research I landed here :crying_cat_face: 

I have multi interfaces nodes and wanted to be full cilium, replacing kube-proxy, kube-router and metallb.

Is there any tracking ticket to support it out of kube-proxy? should it be added as a warning on kube-proxy replacement doc page in the meantime?

Reply:
@n0rad Please open a bug report. This discussion is about getting rid of iptables, but it seems you are looking at getting this to work with our kube-proxy replacement, which is a different thing.